### PR TITLE
fix(select): make floating labels work for ion-select

### PR DESCRIPTION
**Pull request by imgx64**
#### Short description of what this resolves:
This makes floating labels work with ion-select, instead of behaving like stacked labels.

#### Changes proposed in this pull request:
- fix(select): make floating labels work for ion-select

**Ionic Version**: 2.x

**Fixes**: #10751

### CODE REVIEW

File: src/components/label/label.scss [hunk 1]
```diff
@@ -65,7 +65,3 @@ ion-label[floating] {
 
   max-width: 100%;
 }
-
```
Comment (manucorporat) [inline src/components/label/label.scss]: why is this removed?

**Comment by imgx64** [comment]
Once I fixed `input-has-value`, I realized that it made it possible to make floating labels work just by deleting a single CSS rule. So I did just that, and changed the purpose of the change.

I think that CSS rule was added as a workaround because ion-select didn't support `input-has-value`.

File: src/components/label/label.scss (hunk 1 above)
Comment (imgx64) [inline src/components/label/label.scss]: This CSS rule was forcing the floating label to always "float". It was needed because floating labels only float when the ion-item has `input-has-value` class. Now that it's being set by the ion-select, there's no need to force it.

Comment (manucorporat) [inline src/components/label/label.scss]: Fair enough

**Comment by manucorporat** [comment]
Merged! thank you very much for the PR!
